code/__HELPERS/game.dm

```diff
@@ -159,6 +159,37 @@
 		processing_list.Cut(1, 2)
 		processing_list += A.contents
 
+// Recursive loop based off of the recursive mob check. Necessary for preventing organ decay without running every tick.
+// Takes in a variable "atom/O" and looks for any/all organs to toggle decompose, excluding other organ safe containers
+/proc/recursive_organ_check(atom/O, var/decompose)
+
+	var/list/processing_list = list(O)
+	var/list/processed_list = list()
+	var/obj/item/organ/found_organ
+
+	while(processing_list.len)
+
+		var/atom/A = processing_list[1]
+
+		if(istype(A, /obj/item/organ))
+			found_organ = A
+			found_organ.can_decompose = decompose
+
+		else if(istype(A, /mob/living/carbon))
+			var/mob/living/carbon/Q = A
+			for(var/organ in Q.internal_organs)
+				found_organ = organ
+				found_organ.can_decompose = decompose
+
+		for(var/atom/B in A)
+			if(!processed_list[B] && !istype(B, /obj/structure/closet/crate/freezer) && !istype(B, /obj/structure/closet/secure_closet/freezer))
+				processing_list|= B

```

Comment:
If you're already checking the the list contains B with !processed_list[B], don't use |=, it's extremely slow compared to +=

Reply:
can organs contain other organs with this?

Reply:
They should be able to. Mobs don't since it's stored in internal_organs, so they have that snowflake check.